sdk/src/logs/logger.cc

```diff
@@ -30,88 +30,31 @@ const nostd::string_view Logger::GetName() noexcept
-                 nostd::string_view body,
-                 const common::KeyValueIterable &attributes,
-                 trace_api::TraceId trace_id,
-                 trace_api::SpanId span_id,
-                 trace_api::TraceFlags trace_flags,
-                 common::SystemTimestamp timestamp) noexcept
+nostd::unique_ptr<opentelemetry::logs::LogRecord> Logger::CreateLogRecord() noexcept
 {
-  // If this logger does not have a processor, no need to create a log record
+  // If this logger does not have a processor, no need to create a log recordable
   if (!context_)
   {
-    return;
+    return nullptr;
   }
-  auto &processor = context_->GetProcessor();
 
-  // TODO: Sampler (should include check for minSeverity)
+  return nostd::unique_ptr<opentelemetry::logs::LogRecord>(
+      context_->GetProcessor().MakeRecordable().release());
+}
 
-  auto recordable = processor.MakeRecordable();
-  if (recordable == nullptr)
+void Logger::EmitLogRecord(nostd::unique_ptr<opentelemetry::logs::LogRecord> &&log_record) noexcept
+{
+  if (!log_record)
   {
-    OTEL_INTERNAL_LOG_ERROR("[LOGGER] Recordable creation failed");
     return;
   }
 
-  // Populate recordable fields
-  recordable->SetTimestamp(timestamp);
-  recordable->SetSeverity(severity);
-  recordable->SetBody(body);
-  recordable->SetInstrumentationScope(GetInstrumentationScope());
-
-  recordable->SetResource(context_->GetResource());
-
-  attributes.ForEachKeyValue([&](nostd::string_view key, common::AttributeValue value) noexcept {
-    recordable->SetAttribute(key, value);
-    return true;
-  });
-
-  // Inject trace_id/span_id/trace_flags if none is set by user
-  auto provider     = trace_api::Provider::GetTracerProvider();
-  auto tracer       = provider->GetTracer(logger_name_);
-  auto span_context = tracer->GetCurrentSpan()->GetContext();
-
-  // Leave these fields in the recordable empty if neither the passed in values
-  // nor the context values are valid (e.g. the application is not using traces)
-
-  // TraceId
-  if (trace_id.IsValid())
-  {
-    recordable->SetTraceId(trace_id);
-  }
-  else if (span_context.trace_id().IsValid())
-  {
-    recordable->SetTraceId(span_context.trace_id());
-  }
-
-  // SpanId
-  if (span_id.IsValid())
-  {
-    recordable->SetSpanId(span_id);
-  }
-  else if (span_context.span_id().IsValid())
-  {
-    recordable->SetSpanId(span_context.span_id());
-  }
+  auto &processor = context_->GetProcessor();
 
-  // TraceFlags
-  if (trace_flags.IsSampled())
-  {
-    recordable->SetTraceFlags(trace_flags);
-  }
-  else if (span_context.trace_flags().IsSampled())
-  {
-    recordable->SetTraceFlags(span_context.trace_flags());
-  }
+  // TODO: Sampler (should include check for minSeverity)
 
-  // Send the log record to the processor
-  processor.OnEmit(std::move(recordable));
+  // Send the log recordable to the processor
+  processor.OnEmit(std::unique_ptr<Recordable>(static_cast<Recordable *>(log_record.release())));
```

Comment:
Oh I didn't realize this would need conversion from nostd::unique_ptr to std::unique_ptr.

Reply:
Because API create `nostd::unique_ptr` but sdk and exporters use `std::unique_ptr`, as we talked before at https://github.com/open-telemetry/opentelemetry-cpp/pull/1766#discussion_r1037982249_